llama-index-core/llama_index/core/base/base_query_engine.py

```diff
@@ -92,13 +105,16 @@ async def asynthesize(
             "This query engine does not support asynthesize, use aquery directly"
         )
 
-    @abstractmethod
     def _query(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
-        pass
+        return asyncio_run(self._aquery(query_bundle=query_bundle))
 
-    @abstractmethod
     async def _aquery(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
-        pass
+        with self.callback_manager.event(
+            CBEventType.QUERY, payload={EventPayload.QUERY_STR: query_bundle.query_str}
+        ) as query_event:
+            response = await self.run(query_bundle=query_bundle)
```

Comment:
without defining any `@step` methods in the base class, won't `self.run()` not run anything? It requires the subclass to have `@step` methods in order to work?

Reply:
imo here for example, `def _aquery()` should be the step method, and `def aquery()` be the workflow entry point? (I might be wrong)

Reply:
Or I guess, if we rely on `_aquery()` being the entry point, that will be a breaking change

Reply:
to explain my rationale: 
- we want to migrate query engines to workflows
- we don't want to do it all at once. most existing query engines override `_query` and `_aquery` 

if this PR lands, then any query engine that hasn't been migrated over (not retriever or sub-question query engine), will call `_query` or `_aquery` which (ideally) overrides this method on the base class 

this parent method will only run for query engines that have been migrated to workflows